backend/dataall/modules/omics/services/omics_service.py

```diff
@@ -0,0 +1,164 @@
+from dataall.core.permissions.services.tenant_policy_service import TenantPolicyService
+from dataall.core.permissions.services.group_policy_service import GroupPolicyService
+from dataall.modules.s3_datasets.db.dataset_repositories import DatasetRepository
+from dataall.base.db import exceptions
+import json
+
+from dataall.modules.omics.db.omics_repository import OmicsRepository
+from dataall.modules.omics.aws.omics_client import OmicsClient
+from dataall.modules.omics.db.omics_models import OmicsRun
+from dataall.modules.omics.services.omics_permissions import (
+    MANAGE_OMICS_RUNS,
+    CREATE_OMICS_RUN,
+    OMICS_RUN_ALL,
+    DELETE_OMICS_RUN,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class OmicsService:
+    """
+    Encapsulate the logic of interactions with Omics.
+    """
+
+    @staticmethod
+    @TenantPolicyService.has_tenant_permission(MANAGE_OMICS_RUNS)
+    @ResourcePolicyService.has_resource_permission(CREATE_OMICS_RUN)
+    @GroupPolicyService.has_group_permission(CREATE_OMICS_RUN)
+    def create_omics_run(*, uri: str, admin_group: str, data: dict) -> OmicsRun:
+        """
+        Creates an omics_run and attach policies to it
+        Throws an exception if omics_run are not enabled for the environment
+        """
+
+        with _session() as session:
+            environment = EnvironmentService.get_environment_by_uri(session, uri)
+            dataset = DatasetRepository.get_dataset_by_uri(session, data['destination'])
+            enabled = EnvironmentService.get_boolean_env_param(session, environment, 'omicsEnabled')
+            workflow = OmicsRepository(session=session).get_workflow(workflowUri=data['workflowUri'])
+            group = EnvironmentService.get_environment_group(session, admin_group, environment.environmentUri)
+
+            if not enabled and enabled.lower() != 'true':
+                raise exceptions.UnauthorizedOperation(
+                    action=CREATE_OMICS_RUN,
+                    message=f'OMICS_RUN feature is disabled for the environment {environment.label}',
+                )
+
+            omics_run = OmicsRun(
+                owner=get_context().username,
+                organizationUri=environment.organizationUri,
+                environmentUri=environment.environmentUri,
+                SamlAdminGroupName=admin_group,
+                workflowUri=data['workflowUri'],
+                parameterTemplate=data['parameterTemplate'],
+                label=data['label'],
+                outputUri=f's3://{dataset.S3BucketName}',
+                outputDatasetUri=dataset.datasetUri,
+            )
+
+            OmicsRepository(session).save_omics_run(omics_run)
+
+            response = OmicsClient(awsAccountId=environment.AwsAccountId, region=environment.region).run_omics_workflow(
+                omics_workflow=workflow, omics_run=omics_run, role_arn=group.environmentIAMRoleArn
+            )
```

Comment:
Can we switch the order of these 2  so that we first create the run before saving the run to Aurora?

Otherwise - if the run does not get created for some reason (i.e. permission error, invalid input, etc.) then the record still exists in Aurora DB but does not exist on HealthOmics runs tab so getting the run details will fail

Reply:
Tested flipping the order to the following and works as expected with no new RDS record being created in there is an issue starting the run:

```
            response = OmicsClient(awsAccountId=environment.AwsAccountId, region=environment.region).run_omics_workflow(
                omics_workflow=workflow, omics_run=omics_run, role_arn=group.environmentIAMRoleArn
            )

            omics_run.runUri = response['id']
            OmicsRepository(session).save_omics_run(omics_run)
```